### PR TITLE
Apply further optimizations  to avoid memory leaks

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
It looks like `through` module usage causes serious memory leaks. So I've changed the flow to use simple `on('data')` streaming approach

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,13 +51,18 @@ The `exec` function takes some options, with these defaults:
   isBuffer: false,
   shell: undefined,
   logger: undefined,
+  maxStdoutBufferSize: 20 * 1024 * 1024, // 20 MB
+  maxStdoutBufferSize: 20 * 1024 * 1024, // 20 MB
```
Comment (jlipps) [inline README.md]: i don't see any reason to repeat the same key, did you mean `Stderr` perhaps?

File: lib/exec.js [hunk 2]
```diff
@@ -45,28 +48,46 @@ async function exec (cmd, args = [], opts = {}) {
...
+    let stdoutBufferSize = 0;
     if (proc.stdout) {
       proc.stdout.on('error', (err) => {
         reject(new Error(`Standard output '${err.syscall}' error: ${err.stack}`));
       });
       // keep track of stdout if we have not said not to
-      if (!opts.ignoreOutput) {
+      if (opts.ignoreOutput) {
+        // https://github.com/nodejs/node/issues/4236
+        proc.stdout.on('data', () => {});
```
Comment (jlipps) [inline lib/exec.js]: rather than listening to the event with a noop, could we also just ... not listen to the event?

File: lib/exec.js [hunk 3]
```diff
@@ -45,28 +48,46 @@ async function exec (cmd, args = [], opts = {}) {
...
+            stdoutBufferSize -= _.size(_.first(stdoutArr));
+            stdoutArr.shift();
+          }
           if (opts.logger && _.isFunction(opts.logger.debug)) {
             opts.logger.debug(data);
           }
         });
       }
     }
+    let stderrBufferSize = 0;
```
Comment (jlipps) [inline lib/exec.js]: there's starting to be an awful lot of code which is duplicated for both stdout and stderr in this section. does it make sense to refactor for code reuse?

File: lib/exec.js (hunk 2 above)
Comment (mykola-mokhnach) [reply lib/exec.js]: have you checked the bug description above?

Comment (jlipps) [reply lib/exec.js]: nope, makes sense now!

File: README.md (hunk 1 above)
Comment (mykola-mokhnach) [reply README.md]: it is a copypaste issue, good catch

File: lib/exec.js (hunk 3 above)
Comment (mykola-mokhnach) [reply lib/exec.js]: I'll try to do that

File: package.json [hunk 4]
```diff
@@ -34,8 +34,7 @@
     "bluebird": "^3.5.1",
     "lodash": "^4.17.4",
     "shell-quote": "^1.4.3",
-    "source-map-support": "^0.5.3",
-    "through": "^2.3.8"
```
Comment (KazuCocoa) [inline package.json]: 👋

**Pull request merged**